### ISSUE TITLE
Docker and Podman are not enabled out of the box.

### ISSUE THREAD

**Issue by DarkGhostHunter** [ublue-os/bluefin#1739]
### Describe the bug

Docker and Podman are not enabled out of the box. The user needs to set up these manually.

This involves, apart from using `ujust dx-group`:

- Enabling docker: `systemctl enable docker.socket && systemctl start docker.socket` 
- Enabling podman: `systemctl --user enable --now podman.socket`

### What did you expect to happen?

To have Docker and Podman out of the box.

**Comment by p5** [comment]
The Docker socket should start immediately after you call `docker`.  It takes about 0.5 seconds to start.
This change was made [here](https://github.com/ublue-os/bluefin/pull/772), and the reasoning is documented.

We were advised from the Fedora team Podman and Docker can sometimes conflict with eachother, so we only start Docker when the user wants it.  

As for Podman, running without a socket is generally safer, so we default to doing that.

**Comment by p5** [comment]
I've just loaded up a clean Bluefin DX installation, and I can see the docker.socket is enabled and active by default.  When something tries to use that socket, it will start the docker.service.  

Please could you share the output of `systemctl status docker.socket`

**Comment by DarkGhostHunter** [comment]
I get it. When I tried to use it through PHPStorm, Docker and Podman Unix sockets where not found.

Also, when to use Podman or Docker? I know Podman is rootless, but I had compatibility problems with DevContainers so I resorted to Docker. 

There is also [a way to run Docker in rootless mode officially](https://docs.docker.com/engine/security/rootless/).

**Comment by p5** [comment]
Yeah, we use Podman when integrating with the OS (such as terminals, systemd etc) because it's what upstream Fedora recommends, and include Docker for compatibility.  
I tend to default to using Podman since it's more secure, and fallback to Docker if I can't get something working with Podman, but it's really up.to.you which you use and when.

I'm not too familiar with rootless Docker.  I've not tried it myself, so I'm not too sure where that fits in.

**Comment by DarkGhostHunter** [comment]
Well, theoretically should be Podman replacement, but if Fedora uses it upstream I guess we will have to live with that.

Closing this PR.
